### PR TITLE
[Ecosystem] Enable &Signer in script function

### PR DESCRIPTION
**Pull request by areshand**
## Motivation

Please check https://github.com/aptos-labs/aptos-core/issues/547

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?
Yes

(You must have submitted a [signed CLA](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#contributor-license-agreement) that includes your GitHub handle prior to us accepting and landing your work. Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

cargo nextest

### CODE REVIEW

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 1]
```diff
@@ -124,8 +124,14 @@ impl AptosVM {
         let mut signer_param_cnt = 0;
         // find all signer params at the beginning
         for ty in func.parameters.iter() {
-            if matches!(ty, Type::Signer) {
-                signer_param_cnt += 1;
+            match ty {
+                Type::Signer => signer_param_cnt += 1,
+                Type::Reference(inner_type) => {
+                    if matches!(&**inner_type, Type::Signer) {
```
Comment (davidiw) [inline aptos-move/aptos-vm/src/aptos_vm.rs:130]: good code reuse, hahaha

**Review by davidiw — APPROVED** [review]
this is great

**Comment by davidiw** [comment]
/cancel
linting issues

**Comment by areshand** [comment]
> /cancel linting issues

there is a place to be changed at the move resource viewer as well. will send a PR soon

**Comment by areshand** [comment]
flaky docker test. retry land

**Pull request merged**